File: QID-1577-MVApcp6/MVApcp6.r

# clear all variables
rm(list = ls(all = TRUE))
graphics.off()

# load data
x = read.table("carc.txt")
frame = data.frame(x[, 5:7])
colnames(frame) = c("headroom", "rear seat", "trunk space")

m1 = mean(frame[, 1])
m2 = mean(frame[, 2])
m3 = mean(frame[, 3])

# Plot
boxplot(frame, lwd = 1)
title("Boxplot (Car Data)")
lines(c(0.6, 1.4), c(m1, m1), lty = "dotted", lwd = 1.5, col = "red3")
lines(c(1.6, 2.4), c(m2, m2), lty = "dotted", lwd = 1.5, col = "red3")
lines(c(2.6, 3.4), c(m3, m3), lty = "dotted", lwd = 1.5, col = "red3") 

File: QID-1577-MVApcp6/MVApcp6.py
#works on pandas 1.5.2 and matplotlib 3.6.2
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_table("carc.txt", header=None)
df = df.iloc[:,[4,5,6]]

fig, ax = plt.subplots(figsize = (12,10))
ax.boxplot(df, widths = 0.7, labels = ["headroom","rear seat","trunk space"],
            whiskerprops = {'linestyle': '--', 'linewidth': '2'},
            capprops = {'linewidth': '2'}, capwidths = 0.5,
            boxprops = {'linewidth': '2'}, flierprops = {'linewidth':'2.5', 'color':'red'},
            meanline = True, showmeans = True)
ax.tick_params(axis='both', labelsize=25)
plt.title(label = "Boxplot (Car Data)", 
          fontsize = 30, fontweight = "bold", pad = 15)
